source/core/Variants.cpp

```diff
@@ -26,53 +26,76 @@ SDG
 namespace Vireo
 {
 
-bool VariantType::IsNullVariant(VariantTypeRef variant)
+//------------------------------------------------------------
+VariantType::VariantType(TypeManagerRef typeManager)
+    : TypeCommon(typeManager)
 {
-    return variant == nullptr;
+    _isFlat = false;
+    _ownsDefDefData = false;
+    _isMutableValue = true;
+    _hasCustomDefault = false;
+    _topAQSize = TypeManager::HostPointerToAQSize();
+    _aqAlignment = TypeManager::AQAlignment(sizeof(void*));
+    _encoding = kEncoding_Variant;
 }
 
-bool VariantType::IsNullOrEmptyVariant(VariantTypeRef variant)
+NIError VariantType::InitData(void* pData, TypeRef pattern)
 {
-    return variant == nullptr
-        || (variant->_underlyingTypeRef->BitEncoding() == kEncoding_None  // variant's data is empty
-            && IsAttributeMapNullOrEmpty(variant));  // variant's attributes are not set or empty.
-}
+    NIError err = kNIError_Success;
 
-bool VariantType::IsAttributeMapNullOrEmpty(VariantTypeRef variant)
-{
-    return variant->_attributeMap == nullptr || variant->_attributeMap->size() == 0;
+    auto variantValue = static_cast<VariantDataRef *>(pData);
+    if (*variantValue != nullptr) {
```

Comment:
We might want to call InitData on a pointer already initialized to set it to the default value. In that scenario consider instead deleting the current content of pData and then continue with the initialization code.

Reply:
I think InitData should not delete data. It looks like InitData for other types are not designed that way. I have added an assert to find out if we will ever get to the scenario. We can address this if there is a valid test case. This seems to work correctly for variant inside array and cluster correctly. We should refrain from changing this without a test case. Also, please see my detailed comment in ArrayType::InitData. We should address that issue for all the types correctly to avoid the performance issue.